include/private/tdycoreimpl.h

```diff
@@ -55,10 +57,12 @@ struct _TDyOps {
   // before the DM is distributed across processes.
   PetscErrorCode (*set_dm_fields)(void*, DM);
 
+  PetscInt (*get_num_dm_fields)(void*);
```

Comment:
Can you add a comment above here? It's just easier to have everything documented up front.